SEED-platform/seed
# 1946 pm meter import fix

#### Any background context you want to provide?
From the here https://github.com/SEED-platform/seed/issues/1946#issuecomment-579642099
> ...
> There's a notable flaw in the way I built out the importing of PM meters. That is, the PM import feature for meters and meter readings was built on the assumption that PM Property ID would always be at least one of the matching fields. During a PM meter import, I assumed it would be enough to find just one record with a given PM Property ID and then associate meters to that lone record. The thought was that matching and linking would take care of associating the meters and meter readings to multiple records via that first found record.
> 
> Of course, as you both know, PM Property ID may or may not be considered "matching criteria" within any given organization. I think the ideal scenario is _still_ to rely on PM Property ID during PM meter imports, but instead, apply incoming meters and meter readings to **all** records within a given cycle that have the given PM Property ID. I don't think this would be a massive effort, but I wanted to get both of your reactions on this before thinking about implementation steps.
> 
> Note that it would still be the case that meters are cycle-agnostic. Meaning, if a property has meters and meter readings and is linked to a property in another cycle, that other cycle['s PropertyState] will be associated to the same meters and meter readings.

#### What's this PR do?
Updates the PM Meter import code to associate incoming meters and meter readings to **all**  records across an org's cycles with a given PM Property ID.

Updates the feedback provided to the user on import to list the Cycles for each record.

**Caveat**: In cases where a single cycle has multiple properties with the same PM Property ID that's having meters and meter readings imported, it could look like there are duplicate summary rows.
<img width="1124" alt="Screen Shot 2020-03-18 at 10 46 48 AM" src="https://user-images.githubusercontent.com/30608004/76985413-cc350080-6905-11ea-9885-52c52d3e6374.png">
This is a product of it being difficult to distinguish these affected records in this view. From the backend perspective, these are differentiated by property_id, but showing that to users wouldn't be very useful. 

#### How should this be manually tested?
For code review proposes, using the commits to step through the changes is a viable approach. The most dense commit is the first as it contains a refactor as well as a "feature implementation" (sorry 😅 ).

Import Portfolio Manager meters with "PM Property ID"s that are found among multiple linked and unlinked property records. Here are the scenarios that should be tested:
- PM Meter import target records have the same PM Property ID and have links across cycles.
- PM Meter import target records have the same PM Property ID and are within the same cycle.

#### What are the relevant tickets?
#1946 

#### Screenshots (if appropriate)
Here's a case of linked records being PM Meter import targets:
<img width="1122" alt="Screen Shot 2020-03-18 at 10 51 33 AM" src="https://user-images.githubusercontent.com/30608004/76986257-09e65900-6907-11ea-85c8-8f5c8c249c50.png">
